### PR TITLE
Rename OpenMPExec to OpenMPInternal in line with the other backends

### PR DESCRIPTION
**Pull request by Rombur**
This PR renames `OpenMPExec` to `OpenMPInternal`.

Part of https://github.com/kokkos/kokkos/issues/4935

### CODE REVIEW

File: core/src/OpenMP/Kokkos_OpenMP_Instance.cpp [hunk 1]
```diff
@@ -517,7 +515,3 @@ constexpr DeviceType DeviceTypeTraits<OpenMP>::id;
 #endif
 
 }  // namespace Kokkos
-
-#else
-void KOKKOS_CORE_SRC_OPENMP_EXEC_PREVENT_LINK_ERROR() {}
-#endif  // KOKKOS_ENABLE_OPENMP
```
Comment (Rombur) [inline core/src/OpenMP/Kokkos_OpenMP_Instance.cpp:523]: This is the only change except for the renaming. If `KOKKOS_ENABLE_OPENMP` is false, the `OpenMP` directory is not included, so this is not necessary.

File: core/src/OpenMP/Kokkos_OpenMP_Instance.cpp [hunk 2]
```diff
@@ -57,19 +54,20 @@
 #include <impl/Kokkos_Error.hpp>
 #include <impl/Kokkos_CPUDiscovery.hpp>
 #include <impl/Kokkos_Tools.hpp>
+#include "Kokkos_OpenMP_Instance.hpp"
```
Comment (dalg24) [inline core/src/OpenMP/Kokkos_OpenMP_Instance.cpp:57]: Don't we get it transitively from `<Kokkos_OpenMP.hpp>` which is included from `<Kokkos_Core.hpp>`
(I don't necessary have an issue with your explicitly including it)

File: core/src/OpenMP/Kokkos_OpenMP_Instance.hpp [hunk 3]
```diff
@@ -66,28 +64,35 @@
...
+class OpenMPInternal;
 
 extern int g_openmp_hardware_max_threads;
 
 extern __thread int t_openmp_hardware_id;
-extern __thread OpenMPExec* t_openmp_instance;
+extern __thread OpenMPInternal* t_openmp_instance;
+
+struct OpenMPTraits {
+  static int constexpr MAX_THREAD_COUNT = 512;
```
Comment (dalg24) [inline core/src/OpenMP/Kokkos_OpenMP_Instance.hpp:78]: OK

**Review by dalg24 — APPROVED** [review]

**Review by crtrott — APPROVED** [review]

File: core/src/OpenMP/Kokkos_OpenMP_Instance.cpp (hunk 2 above)
Comment (Rombur) [reply core/src/OpenMP/Kokkos_OpenMP_Instance.cpp:57]: Yes, you do. It's just the habit of adding the `.hpp` with the corresponding `.cpp`.

**Review by masterleinad — APPROVED** [review]

**Comment by dalg24** [comment]
Failure (OpenMPTarget on NVIDIA GPU) is unrelated
```
4: [ RUN      ] openmptarget.reducers_struct
4: /var/jenkins/workspace/Kokkos/core/unit_test/TestReducers.hpp:320: Failure
4: Expected equality of these values:
4:   sum_scalar
4:     Which is: 32-byte object <00-16 44-47 00-16 44-47 9F-C1 8C-FE 00-60 E7-44 00-16 44-47 00-16 44-47 00-16 44-47 00-16 44-47>
4:   reference_sum
4:     Which is: 32-byte object <00-16 44-47 00-16 44-47 00-16 44-47 00-16 44-47 00-16 44-47 00-16 44-47 00-16 44-47 00-16 44-47>
4: [  FAILED  ] openmptarget.reducers_struct (3 ms)
```
Probably another manifestation of #5025

**Pull request merged**